### ISSUE TITLE
{}[Style]: Failed to load sprite: CGImageSourceCreateImageAtIndex failed

### ISSUE THREAD

**Issue by always-beta** [mapbox/mapbox-gl-native#11310]
SDK Version: iOS v3.6.3

A few of our app users (both ios and android) reported that there was no poi display, and after research, it seems that the cache of sprite which is stored in the cache.db is invalid, I tried to remove the sprite@2x.json and sprite@2x.png from cache.db and it works properly.

I found following log

`{}[Style]: Failed to load sprite: CGImageSourceCreateImageAtIndex failed`

from ios whenever a new map instance is created.

I failed to find a reproduce step. So I exported **data** field of sprite@2x.json and sprite@2x.png file from cache.db's **resources** table and attached. Both of them have value of  1 in the **compressed** field. (Not sure what compression it was used so can't check if the files are valid myself). And still have the correct unchanged etag value as the server provide in **etag** field.

[sprite@2x.zip](https://github.com/mapbox/mapbox-gl-native/files/1757620/sprite.2x.zip)

The sprite@2x.png should be around 400kb but is 3k in db.

Is there anyway I can check what was written to the **data** field originally? And where should I look into to locate the issue?

**Comment by kkaefer** [comment]
The data is zlib/deflate compressed. When uncompressing it, I get a captive portal of a Korean Wifi hotspot. This happens when some captive portals redirect unencrypted requests to other pages until the user has logged in/paid/agreed to TOS. Mapbox GL doesn't have protection against this kind of MITM attack. Instead, you should rely on SSL/TLS encryption to verify resource integrity.

**Comment by kkaefer** [comment]
Related: https://github.com/mapbox/mapbox-gl-native/issues/9225

**Comment by always-beta** [comment]
Thanks kkaefer! 

For those of who encounter same issue with me, my temp fix was:

Check resources table every app launch, select sprite json and sprite image resources, parsing them to see if they can parse to correct file format successfully, if can't, just remove them.

Other kind of resources should have the same problem, but since sprite json and image caused me the most serious problem so I decided to deal with them first.

It works fine, and I checked how many users end up with removing invalid resources on production env, and it turned out way much more than I thought (maybe because I'm targeting users who travel around the world and they tend to have less stable network), so I think devs would be better to pay attention if  load resources using HTTP.
